File: library/src/test/java/org/crispy_fish/datatype/underscore_pairs/SimpleStringUnderscorePairReaderTest.java
package org.crispy_fish.datatype.underscore_pairs;

import org.coner.crispy_fish.datatype.underscore_pairs.SimpleStringUnderscorePairReader;
import org.junit.*;

import static org.assertj.core.api.Assertions.assertThat;

public class SimpleStringUnderscorePairReaderTest {

    private static final String PAIRS_EXAMPLE_BASIC = "key1_value1_key2_value2_key3_value3";

    private SimpleStringUnderscorePairReader reader;

    @Before
    public void setup() {
        reader = new SimpleStringUnderscorePairReader();
    }

    @Test
    public void whenGetFirstKeyItShouldReturnCorrectValue() throws Exception {
        String actual = reader.get(PAIRS_EXAMPLE_BASIC, "key1");

        assertThat(actual).isEqualTo("value1");
    }

    @Test
    public void whenGetInnerKeyItShouldReturnCorrectValue() throws Exception {
        String actual = reader.get(PAIRS_EXAMPLE_BASIC, "key2");

        assertThat(actual).isEqualTo("value2");
    }

    @Test
    public void whenGetLastKeyItShouldReturnCorrectValue() throws Exception {
        String actual = reader.get(PAIRS_EXAMPLE_BASIC, "key3");

        assertThat(actual).isEqualTo("value3");
    }

    @Test
    public void whenGetKeyDoesNotExistItShouldReturnNull() throws Exception {
        String actual = reader.get(PAIRS_EXAMPLE_BASIC, "absent");

        assertThat(actual).isNull();
    }
}

File: library/src/test/kotlin/org/coner/crispy_fish/datatype/underscore_pairs/SimpleStringUnderscorePairReaderTest.kt
package org.coner.crispy_fish.datatype.underscore_pairs

import org.junit.*

import org.assertj.core.api.Assertions.assertThat

class SimpleStringUnderscorePairReaderTest {

    private lateinit var reader: SimpleStringUnderscorePairReader

    private val PAIRS_EXAMPLE_BASIC = "key1_value1_key2_value2_key3_value3"

    @Before
    fun setup() {
        reader = SimpleStringUnderscorePairReader()
    }

    @Test
    fun whenGetFirstKeyItShouldReturnCorrectValue() {
        val actual = reader.get(PAIRS_EXAMPLE_BASIC, "key1")

        assertThat(actual).isEqualTo("value1")
    }

    @Test
    fun whenGetInnerKeyItShouldReturnCorrectValue() {
        val actual = reader.get(PAIRS_EXAMPLE_BASIC, "key2")

        assertThat(actual).isEqualTo("value2")
    }

    @Test
    fun whenGetLastKeyItShouldReturnCorrectValue() {
        val actual = reader.get(PAIRS_EXAMPLE_BASIC, "key3")

        assertThat(actual).isEqualTo("value3")
    }

    @Test
    fun whenGetKeyDoesNotExistItShouldReturnNull() {
        val actual = reader.get(PAIRS_EXAMPLE_BASIC, "absent")

        assertThat(actual).isNull()
    }

}
